bigquery/snippets/browse_table.rb

```diff
@@ -11,23 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-# [START bigquery_quickstart]
-# Imports the Google Cloud client library
+# [START bigquery_browse_table]
 require "google/cloud/bigquery"
 
-# Your Google Cloud Platform project ID
-project_id = "YOUR_PROJECT_ID"
-
-# Instantiates a client
-bigquery = Google::Cloud::Bigquery.new project: project_id
-
-# The name for the new dataset
-dataset_name = "my_new_dataset"
+def browse_table
+  bigquery = Google::Cloud::Bigquery.new project_id: "bigquery-public-data"
```

Comment:
Yes, this is the only way to do it for now. I filed [an issue](https://github.com/googleapis/google-cloud-ruby/issues/2609) because this behavior makes it impossible to run non-free operations on public tables, and when that is solved I'll be able to update this snippet as well.